addons/survey/controllers/main.py

```diff
@@ -321,6 +352,23 @@ def survey_begin(self, survey_token, answer_token, **post):
         })
         return self._prepare_question_html(survey_sudo, answer_sudo, **post)
 
+    @http.route('/survey/next_question/<string:survey_token>/<string:answer_token>', type='json', auth='public', website=True)
+    def survey_next_question(self, survey_token, answer_token, **post):
+        """ Method used to display the next survey question in an ongoing session.
+        Triggered on all attendees screens when the host goes to the next question. """
+        access_data = self._get_access_data(survey_token, answer_token, ensure_token=True)
+        if access_data['validity_code'] is not True:
+            return {'error': access_data['validity_code']}
+        survey_sudo, answer_sudo = access_data['survey_sudo'], access_data['answer_sudo']
+
+        if answer_sudo.state == 'new' and answer_sudo.is_session_answer:
+            answer_sudo.write({
```

Comment:
I vote to introduce a _mark_in_progress(self, **vals) and use it here + in /submit and /begin (or remove the update in /submit as it seems you think dbe thinks qmo suggested it was not necessary, maybe).

Reply:
Done without `**vals` as is did not seem necessary for now (and is not available in 'mark_done' either).